algoliasearch/type_synonym.go

```diff
@@ -0,0 +1,135 @@
+package algoliasearch
+
+import "fmt"
+
+type Synonyms struct {
+	AltCorrectionSynonyms []AltCorrectionSynonym
+	OneWaySynonyms        []OneWaySynonym
+	PlaceholderSynonyms   []PlaceholderSynonym
+	SimpleSynonyms        []SimpleSynonym
+}
+
+type Synonym interface{}
+
+type AltCorrectionSynonym struct {
+	Corrections       []string               `json:"corrections"`
+	HighlightedResult map[string]interface{} `json:"_highlightedResult"`
+	ObjectID          string                 `json:"objectID"`
+	Type              string                 `json:"type"`
+	Word              string                 `json:"word"`
+}
+
+type OneWaySynonym struct {
+	HighlightedResult map[string]interface{} `json:"_highlightedResult"`
+	Input             string                 `json:"input"`
+	ObjectID          string                 `json:"objectID"`
+	Synonyms          []string               `json:"synonyms"`
+	Type              string                 `json:"type"`
+}
+
+type PlaceholderSynonym struct {
+	HighlightedResult map[string]interface{} `json:"_highlightedResult"`
+	ObjectID          string                 `json:"objectID"`
+	Placeholder       string                 `json:"placeholder"`
+	Replacements      []string               `json:"replacements"`
+	Type              string                 `json:"type"`
+}
+
+type SimpleSynonym struct {
+	HighlightedResult map[string]interface{} `json:"_highlightedResult"`
+	ObjectID          string                 `json:"objectID"`
+	Synonyms          []string               `json:"synonyms"`
+	Type              string                 `json:"type"`
+}
+
+func generateSynonyms(rawHits interface{}) (synonyms Synonyms, err error) {
+	hits, ok := rawHits.([]interface{})
```

Comment:
Because where I use the `generateSynonyms`, the parameter comes from a `map[string]interface{}`. Look at `Index.SearchSynonyms`.